hal/src/b5som/network/quectel_ncp_client.cpp

```diff
@@ -492,7 +524,21 @@ int QuectelNcpClient::getCellularGlobalIdentity(CellularGlobalIdentity* cgi) {
     CHECK_TRUE(connState_ != NcpConnectionState::DISCONNECTED, SYSTEM_ERROR_INVALID_STATE);
     CHECK_TRUE(cgi, SYSTEM_ERROR_INVALID_ARGUMENT);
     CHECK(checkParser());
-    CHECK(queryAndParseAtCops(nullptr));
+
+    // FIXME: this is a workaround for CH28408
+    CellularSignalQuality qual;
+    CHECK(queryAndParseAtCops(&qual));
+    CHECK_TRUE(qual.accessTechnology() != CellularAccessTechnology::NONE, SYSTEM_ERROR_INVALID_STATE);
+    // Update current RAT
+    act_ = qual.accessTechnology();
+    // Invalidate LAC and Cell ID
+    cgi_.location_area_code = std::numeric_limits<LacType>::max();
+    cgi_.cell_id = std::numeric_limits<CidType>::max();
+    // Fill in LAC and Cell ID based on current RAT, prefer PSD and EPS
```

Comment:
In case we did not get the LAC and CID from the At commands (for any reason), we will be sending the max values to console. Is that something we need to aware the device-service with......?

Reply:
That's an error value, which is handled in `cellular_global_identity()` resulting in `SYSTEM_ERROR_BAD_DATA` sent in vitals, which we've seen as `-280` errors.

Reply:
Max will look like `0xFFFFFFF`? 
But that is also a valid cell id as per 
https://en.wikipedia.org/wiki/GSM_Cell_ID

Reply:
Max is 0xffffffff, which is not a valid cell id, as it's 2^32 -1.

> A valid CID ranges from 0 to 65535 (2^16 − 1) on GSM and CDMA networks and from 0 to 268,435,455 (2^28 − 1) on UMTS and LTE networks.

Reply:
Perfect, thanks